components/experimental/src/measure/parser.rs

```diff
@@ -222,6 +223,17 @@ impl MeasureUnitParser {
             return Err(InvalidUnitError);
         }
 
+        let single_units = match single_units.len() {
+            0 => SingleUnitVec::Zero,
+            1 => SingleUnitVec::One(single_units.remove(0)),
+            2 => SingleUnitVec::Two(single_units.remove(0), single_units.remove(0)),
+            #[cfg(feature = "alloc")]
+            _ => SingleUnitVec::Multi(single_units.into_vec()),
+
+            #[cfg(not(feature = "alloc"))]
+            _ => return Err(InvalidUnitError),
+        };
```

Comment:
Suggestion: this looks like something that should be a helper fn on SingleUnitVec

Reply:
done